miio/chuangmi_plug.py

```diff
@@ -0,0 +1,188 @@
+             self.load_power,
+             self.wifi_led)
+        return s
+
+
+class ChuangmiPlug(Device):
+    """Main class representing the Chuangmi Plug V1 and V3."""
+
+    def __init__(self, ip: str = None, token: str = None, start_id: int = 0,
+                 debug: int = 0, lazy_discover: bool = True,
+                 model: str = MODEL_CHUANGMI_PLUG_M1) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_CHUANGMI_PLUG_M1
+
+    def status(self) -> ChuangmiPlugStatus:
+        """Retrieve properties."""
+        properties = AVAILABLE_PROPERTIES[self.model]
+        values = self.send(
+            "get_prop",
+            properties
+        )
+
+        properties_count = len(properties)
+        values_count = len(values)
+        if properties_count != values_count:
+            _LOGGER.debug(
+                "Count (%s) of requested properties does not match the "
+                "count (%s) of received values.",
+                properties_count, values_count)
+
+        if self.model == MODEL_CHUANGMI_PLUG_V3:
+            load_power = self.send("get_power", [])  # Response: [300]
+            if len(load_power) == 1:
+                properties.append('load_power')
+                values.append(load_power[0])
+
+        return ChuangmiPlugStatus(
+            defaultdict(lambda: None, zip(properties, values)))
+
+    def on(self):
+        """Power on."""
+        if self.model == MODEL_CHUANGMI_PLUG_V1:
+            return self.send("set_on", [])
+
+        return self.send("set_power", ["on"])
+
+    def off(self):
+        """Power off."""
+        if self.model == MODEL_CHUANGMI_PLUG_V1:
+            return self.send("set_off", [])
+
+        return self.send("set_power", ["off"])
+
+    def usb_on(self):
+        """Power on."""
+        return self.send("set_usb_on", [])
+
+    def usb_off(self):
+        """Power off."""
+        return self.send("set_usb_off", [])
+
+    def set_wifi_led(self, led: bool):
+        """Set the wifi led on/off."""
+        if led:
+            return self.send("set_wifi_led", ["on"])
+        else:
+            return self.send("set_wifi_led", ["off"])
+
+
+class ClassDeprecatedPlug(ChuangmiPlug):
+    def __init__(self, ip: str = None, token: str = None, start_id: int = 0,
+                 debug: int = 0, lazy_discover: bool = True,
+                 model: str = MODEL_CHUANGMI_PLUG_M1) -> None:
+        warnings.simplefilter('always', DeprecationWarning)
+        warnings.warn(
+            "This device class is deprecated. Please use the ChuangmiPlug "
+            "in future and select a model by parameter 'model'.",
+            DeprecationWarning, stacklevel=2)
+
+        super().__init__(ip, token, start_id, debug, lazy_discover,
+                         model=model)
+
+
+class Plug(ClassDeprecatedPlug):
```

Comment:
I created PR #276 to move the deprecated decorator out from vacuumcontainers, it should be used to decorate these classes to inform anyone trying to initialize these classes.